src/components/forms/TextInput.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { throttle, debounce } from 'throttle-debounce';
+import IconInput from 'components/forms/IconInput';
+import DecoratedInput from 'components/forms/DecoratedInput';
+
+export const throttleValue = 500;
+
+const TextInput = ({
+  label,
+  errorText,
+  throttleOnChange,
+  required,
```

Comment:
Interesting... 

So what I've been doing personally, is if the boolean is spread directly onto the component as expected, even if it causes additional side effects, I keep it that name. So like... a perfect example is the disabled attribute of button. Technically, anchors don't have a disabled prop, but I figured it made more sense to accommodate for the `button` element that does.

In this change, if input itself uses the required prop as it normally does, I'd at least "historically" choose to keep calling it required, so that the developer knows that the prop carries the same intention as what the normal input would. That being "said", I can see the alternative angle of using the `is` prefix the moment it's being used in any capacity beyond its normal usage.

Essentially, I don't feel our usage guidelines totally encapsulates this specific use case.

All that being said, that's just what I've done historically, and honestly either way would make sense to me. What do you think makes the most sense in this situation?

Reply:
The other side thing to note: funny that you mention it, but I technically did "not" add the required field to the input yet. Are we comfortable with the UI that makes the input border completely red if it's a required field? Because that's what happens to our current input.

Reply:
<img width="1115" alt="Screen Shot 2020-03-02 at 1 15 24 PM" src="https://user-images.githubusercontent.com/52427513/75704717-e9bc6600-5c87-11ea-9edd-514b210ddc5f.png">

<img width="1703" alt="Screen Shot 2020-03-02 at 1 16 00 PM" src="https://user-images.githubusercontent.com/52427513/75704769-ffca2680-5c87-11ea-869f-0af18a45f693.png">

So if required is applied to input, this is how it would look.

Reply:
I suspect that this is the `:invalid` state (required but empty). Does it only happen on blur? If so, it might be okay...

Reply:
@akdetrick unfortunately I believe this happens regardless of blur.